tests/test_regression_against_cases.py

```diff
@@ -46,3 +47,27 @@ def test_regression_against_case_with_update(case: Path):
 
     dataset, reference_dataset = xr.align(dataset, reference_dataset)
     assert_allclose(dataset, reference_dataset)
+
+
+@pytest.mark.regression
+@pytest.mark.filterwarnings("ignore:Not all input parameters were used")
+def test_regression_against_case_with_repeated_update():
+    case = CASES_DIR / "SPARC_PRD" / "input.yaml"
+
+    input_parameters, algorithm, _, _ = read_case(case)
+    input_parameters["average_electron_density"] = input_parameters["average_electron_density"][::5]
+    input_parameters["average_electron_temp"] = input_parameters["average_electron_temp"][::5]
+
+    dataset = xr.Dataset(input_parameters)
+
+    first_run = algorithm.update_dataset(dataset)
+    second_run = algorithm.update_dataset(first_run)
+
+    for variable in ["atomic_data"]:
+        first_run = first_run.drop_vars(variable)
+        second_run = second_run.drop_vars(variable)
+
+    assert_allclose(
+        first_run.transpose("dim_average_electron_density", "dim_average_electron_temp", "dim_species", "dim_rho"),
+        second_run.transpose("dim_average_electron_density", "dim_average_electron_temp", "dim_species", "dim_rho"),
+    )
```

Comment:
Can you explain why the `transpose` is here? Why is a simple `assert_equal` of the two results not enough?

Reply:
Not sure — I think the datasets should automatically order themselves to match the dimension ordering. However, this doesn't seem to be working, so I'm manually forcing them to have the same dimension ordering.

I've added a comment and tidied up this code.